packages/headless/src/controllers/knowledge/generated-answer/headless-answerapi-generated-answer.ts

```diff
@@ -119,8 +142,13 @@ const subscribeToSearchRequest = (
     }
 
     lastTriggerParams = triggerParams;
+    // Every time a new search request is triggered, we re-calculate the answer query params
+    const cachedAnswerParams = constructAnswerQueryParams(
+      state,
+      engine.navigatorContext || defaultNodeJSNavigatorContextProvider
+    );
```

Comment:
This could be encapsulated inside `triggerSearchRequest`  since these are two steps that always need to happen together.

triggerSearchRequest can access the state, like mentioned in an earlier comment and call `constructAnswerQueryParams` to build the param from the state.

out of scope of this PR, but I'm not a fan of the naming of `triggerSearchRequest `.

Reply:
I was thinking of creating a ticket to refactor some of the stuff there. I mostly wanted to keep the scope to fixing the bug but I do believe some improvements could be done here.